### PR TITLE
added test case descriptions and hardware matrix suggestions

### PR DESCRIPTION
**Pull request by kl3mdeb**

### CODE REVIEW

File: docs/ecosystem/testing-infrastructure/memtest.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+1. Power on the DUT.
+2. Wait for boot until `BOOT_MENU_STRING` appears.
+3. Press `BOOT_MENU_KEY` to enter the boot menu.
+4. Select the key with a proper number for `Payload [memtest]`.
+5. Check if `Memtest86+` is available.
+
+**Expected result**
+
+The `Memtest86+` is visible at the top of the output.
```
Comment (sulewskiprzemyslaw) [inline docs/ecosystem/testing-infrastructure/memtest.md:56]: I think that we should change that description and write about entering the `Memtest86+` option.

File: docs/ecosystem/testing-infrastructure/memtest.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+4. Select the key with a proper number for `BOOT_MENU_ENTRY`.
+5. Press the `l` key.
+6. Check if `l` refreshes output.
+7. Press the `L` key.
+8. Check if `L` refreshes output.
+
+**Expected result**
+
+The `Memtest86+` is visible before and after pressing `l` and
+`L`.
```
Comment (sulewskiprzemyslaw) [inline docs/ecosystem/testing-infrastructure/memtest.md:115]: I think, that we should split that test case into two independent test cases.

File: docs/ecosystem/testing-infrastructure/memtest.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+1. Power on the DUT.
+2. Wait for boot until `BOOT_MENU_STRING` appears.
+3. Press `BOOT_MENU_KEY` to enter the boot menu.
+4. Select the key with a proper number for `Payload [memtest]`.
+5. Make sure that `State: - Running...` is available.
+6. Wait until `** Pass complete, no errors, press Esc to exit **` appears.
+
+**Expected result**
+
+After some time depending on DUT RAM memory capacity the
```
Comment (sulewskiprzemyslaw) [inline docs/ecosystem/testing-infrastructure/memtest.md:141]: At that point, we should write some about memtes86+ passing conditions

**Review by sulewskiprzemyslaw — COMMENTED** [review]
@kl3mdeb Review is done. Also, I think, that we should move the test cases description to new folder, which we should name dasharo-compatibility-specification

**Comment by kl3mdeb** [comment]
Added after review improvements.

File: docs/ecosystem/testing-infrastructure/dasharo-compatibility-specification/hardware-matrix.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+|                            | 2. Goldpin cables + qspimux (RTE <-> flash memory connection)| `BIOS` slot                                |
+
+## Board No. 2
+
+| Component                  | Description                                                          | Board connection                           |
+|----------------------------|----------------------------------------------------------------------|--------------------------------------------|
+| CPU:                       | AMD Opteron(TM) Processor 6282 SE                                    | `CPU1` slot                                |
+|                            | CPU Cooler                                                           | `CPU_FAN1` connector                       |
+| RAM                        | DDR3 16GB Kingston KVR16R11D4/16                                     | `DIMM_B2` slot                             |
+| Flash memory               | Winbond W25Q64FV 8MB                                                 | `BIOS` slot                                |
```
Comment (miczyg1) [inline docs/ecosystem/testing-infrastructure/dasharo-compatibility-specification/hardware-matrix.md:29]:
```suggestion
| Flash memory               | Winbond W25Q128JV 16MB                                                 | `BIOS` slot                                |
```

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**